better porting entry for key bool fix (#69948)
ansible/ansible: docs/docsite/rst/porting_guides/porting_guide_2.10.rst

* Fixed a bug on boolean keywords that made random strings return 'False', now they should return an error if they are not a proper boolean
  Example: `diff: yes-` was returning `False`.

>   This applies for every keyword that would take a boolean as a value (check_mode, no_log, become, run_once, ...), but not conditionals (`*when:`).

Reviewer: ```suggestion
  This applies for every keyword that would take a boolean as a value (check_mode, no_log, become, run_once, ...), but not conditionals (``*when:``).
```

>   Some playbooks would have seemed to be working as random strings would evaluate as `True`.

Reviewer: ```suggestion
  Some playbooks would have seemed to be working as random strings would evaluate as ``True``.
```

.. code::

      - debug:
          copy: src=test.txt dest=/path/test.txt
        diff: not dont_show_diff
        vars:
          dont_show_diff: false

>   This looked like the value was being templated, but this was not the case. It evaluated to `True` not because the expression `not dont_show_diff` would be so in Jinja, instead it was due to a 'non empty string', which is 'truthy' in Python and Jinja.  The expression `diff: dont_show_diff` would also be `True` previouslly.

Reviewer: ```suggestion
  This looked like the value was being templated, but this was not the case. It evaluated to ``True`` not because the expression ``not dont_show_diff`` was evaluated by Jinja, but instead due to a 'non-empty string', which is 'truthy' in Python and Jinja.  The expression ``diff: dont_show_diff`` would also be ``True`` previously.
```

.. note::

>     When there is an error templating with `no_log` it defaults to `True`, to be on the safe side, so this might obscure the issue is due to this change.

Reviewer: ```suggestion
    When there is an error templating with ``no_log`` it defaults to ``True`` to be on the safe side, so this might obscure that real the issue is due to this change.
```

          dont_show_diff: false

> * The ``no_log`` keyword has been modified to act in a more secure manner, if there is a value assigned and a templating or type conversion error happens, it defaults to ``True`` instead of it' snormal ``False``. This is done to avoid accidental disclosure when a user impoperly sets a template or mispells the boolean values. This might alter the behaviour of playbooks that seemingly 'worked' but were not really processing the option correctly.

Reviewer 2: typo: `it' snormal`
